packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -59,21 +62,19 @@ export const isValidCodeWhispererCoreConnection = (conn?: Connection): conn is C
         return isIamConnection(conn)
     }
 
-    if (isSageMaker()) {
-        return isIamConnection(conn)
-    }
-
     return (
+        isValidSageMakerConnection(conn) ||
         (isCloud9('codecatalyst') && isIamConnection(conn)) ||
         (isSsoConnection(conn) && hasScopes(conn, codeWhispererCoreScopes))
     )
 }
 /** Superset that includes all of CodeWhisperer + Amazon Q */
 export const isValidAmazonQConnection = (conn?: Connection): conn is Connection => {
     return (
-        (isSsoConnection(conn) || isBuilderIdConnection(conn)) &&
-        isValidCodeWhispererCoreConnection(conn) &&
-        hasScopes(conn, amazonQScopes)
+        isValidSageMakerConnection(conn) ||
+        ((isSsoConnection(conn) || isBuilderIdConnection(conn)) &&
+            isValidCodeWhispererCoreConnection(conn) &&
+            hasScopes(conn, amazonQScopes))
```

Comment:
`isValidSageMakerConnection` returns true if it is an IAM or SSO connection only - it doesn't seem to do any checks about sagemaker. That isn't a valid Q connection, we have to do a scope check.
 This means this whole block will return if the connection is not undefined for anyone running the extension, not just in sagemaker.

Based on the current implementation of `isValidCodeWhispererCoreConnection`, this should work:
```suggestion
        isValidCodeWhispererCoreConnection(conn) && hasScopes(conn, amazonQScopes)
```

Reply:
The Sagemaker IAM connection doesn't have amazonQScopes set, it follows a different path from the traditional login flow as it gets loaded through credentials provider.

I attempted to limit the scope for the Sso connection for that check before and received push back on it. I can bring it back here, but it will be specifically for Sso rather than both Sso and IAM